packages/babel-plugin-transform-classes/src/transformClass.js

```diff
@@ -201,6 +206,24 @@ export default function transformClass(
     }
   }
 
+  function ensureClassRefValidInChildScopes(path: NodePath) {
+    path.traverse({
+      Scopable(path) {
+        if (
+          path.type === "ClassDeclaration" ||
+          path.type === "ClassExpression"
+        ) {
+          path.skip();
+          return;
+        }
+
+        if (path.scope.hasOwnBinding(classState.classRef.name)) {
```

Comment:
Copied from https://github.com/babel/babel/blob/17d62c3743ca7c529bc14ddcb890a6a31ed956b1/packages/babel-plugin-transform-classes/src/transformClass.js#L499,L502


```ts
  /**
   * Replace the constructor body of our class.
   */
  function pushConstructor(
    superReturns,
    method: { type: "ClassMethod" },
    path: NodePath,
  ) {
    // https://github.com/babel/babel/issues/1077
    if (path.scope.hasOwnBinding(classState.classRef.name)) {
      path.scope.rename(classState.classRef.name);
    }
    // ...
  }
```

Reply:
We need to traverse here because `super` may appear anywhere in methods.